src/anyio/_core/_tempfile.py

```diff
@@ -0,0 +1,600 @@
+from __future__ import annotations
+
+import os
+import sys
+import tempfile
+from collections.abc import Iterable
+from io import BytesIO, TextIOWrapper
+from types import TracebackType
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    AnyStr,
```

Comment:
`AnyStr` seems to be deprecated (https://github.com/python/cpython/issues/105578), maybe it's a good idea to declare a local type variable like this, to avoid extra code churn in Python 3.16
```py
AnyStr = TypeVar("AnyStr", str, bytes)
```
(`TypeVar` itself isn't being deprecated any time soon even with the new generic syntax)

Reply:
I frankly don't see how that helps with anything. We'll of course switch to the new typevar syntax as soon as it becomes possible, but until then I don't think there's any reason whatsoever to shun `AnyStr`.